libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/BabbageFeatures.hs

```diff
@@ -0,0 +1,303 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Test.Cardano.Ledger.Examples.BabbageFeatures where
+
+import qualified Cardano.Crypto.Hash as CH
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.Alonzo.Data (Data (..), dataToBinaryData)
+import Cardano.Ledger.Alonzo.Language (Language (..))
+import Cardano.Ledger.Alonzo.Scripts (ExUnits (..))
+import qualified Cardano.Ledger.Alonzo.Scripts as Tag (Tag (..))
+import Cardano.Ledger.Alonzo.TxWitness (RdmrPtr (..), Redeemers (..))
+import qualified Cardano.Ledger.Babbage.TxBody as Babbage
+import Cardano.Ledger.BaseTypes
+  ( Network (..),
+    StrictMaybe (..),
+    mkTxIxPartial,
+  )
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Core (EraRule)
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Credential
+  ( Credential (..),
+    StakeReference (..),
+  )
+import qualified Cardano.Ledger.Crypto as CC
+import Cardano.Ledger.Era (Era (..), ValidateScript (hashScript))
+import Cardano.Ledger.Keys
+  ( KeyPair (..),
+    KeyRole (..),
+    hashKey,
+  )
+import Cardano.Ledger.Pretty.Babbage ()
+import Cardano.Ledger.SafeHash (hashAnnotated)
+import Cardano.Ledger.Shelley.API (ProtVer (..), UTxO (..))
+import Cardano.Ledger.Shelley.LedgerState (UTxOState (..), smartUTxOState)
+import Cardano.Ledger.Shelley.UTxO (makeWitnessVKey)
+import Cardano.Ledger.TxIn (TxIn (..))
+import Cardano.Ledger.Val (inject)
+import Control.State.Transition.Extended hiding (Assertion)
+import qualified Data.Compact.SplitMap as SplitMap
+import Data.Default.Class (Default (..))
+import qualified Data.Map as Map
+import GHC.Stack
+import qualified Plutus.V1.Ledger.Api as Plutus
+import Test.Cardano.Ledger.Examples.TwoPhaseValidation
+  ( Expect (..),
+    expectedUTxO,
+    freeCostModel,
+    testUTXOW,
+    trustMeP,
+  )
+import Test.Cardano.Ledger.Generic.Fields
+  ( PParamsField (..),
+    TxBodyField (..),
+    TxField (..),
+    TxOutField (..),
+    WitnessesField (..),
+  )
+import Test.Cardano.Ledger.Generic.PrettyCore ()
+import Test.Cardano.Ledger.Generic.Proof
+import Test.Cardano.Ledger.Generic.Scriptic (PostShelley, Scriptic (..))
+import Test.Cardano.Ledger.Generic.Updaters
+import Test.Cardano.Ledger.Shelley.Generator.EraGen (genesisId)
+import Test.Cardano.Ledger.Shelley.Utils (RawSeed (..), mkKeyPair)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (Assertion, testCase)
+
+-- =======================
+-- Setup the initial state
+-- =======================
+
+scriptAddr :: forall era. (Scriptic era) => Core.Script era -> Proof era -> Addr (Crypto era)
+scriptAddr s _pf = Addr Testnet pCred sCred
+  where
+    pCred = ScriptHashObj . hashScript @era $ s
+    (_ssk, svk) = mkKeyPair @(Crypto era) (RawSeed 0 0 0 0 0)
+    sCred = StakeRefBase . KeyHashObj . hashKey $ svk
```

Comment:
I think it is good practice to put the (Proof era) as the first argument.

Reply:
yes, I agree, thanks for catching that!